File: app-python/entitlements_app/services/get_feature_query_service.py
"""Return one registered feature by feature key."""

from decimal import Decimal

from entitlements_app.requests.get_feature_query_request import GetFeatureQueryRequest
from entitlements_app.entities import Feature
from entitlements_app.ports import EntitlementsRepository


class GetFeatureQueryService:
    """Service handling Query `GetFeature`."""

    def __init__(self, entitlements_repository: EntitlementsRepository) -> None:
        self.entitlements_repository: EntitlementsRepository = entitlements_repository

    def handle(self, request: GetFeatureQueryRequest) -> Feature:
        # spec-driven body for Query GetFeature.
        # intent: Return one registered feature by feature key.
        # reads: EntitlementsRepository.findFeature via `feature`.
        return self.entitlements_repository.findFeature(request.featureKey)

File: app/src/main/java/com/example/spring/GetFeatureQueryService.java
package com.example.spring;

import com.example.spring.error.NotFoundException;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

/**
 * Return one registered feature by feature key.
 */
@Service
public class GetFeatureQueryService {

    private final EntitlementsRepository entitlementsRepository;

    public GetFeatureQueryService(EntitlementsRepository entitlementsRepository) {
        this.entitlementsRepository = entitlementsRepository;
    }

    @Transactional(readOnly = true)
    public Feature handle(GetFeatureQueryRequest request) {
        // === GEN_REGION_START: service-body ===
        // spec-driven body for Query GetFeature.
        // intent: Return one registered feature by feature key.
        // reads: EntitlementsRepository.findFeature via `feature`.
        Feature row = this.entitlementsRepository.findFeature(request.featureKey());
        if (row == null) {
            throw new NotFoundException("Feature not found.");
        }
        return row;
        // === GEN_REGION_END: service-body ===
    }
}
